src/algebra/category/Module/biproducts.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import algebra.group.pi
+import category_theory.limits.shapes.biproducts
+import algebra.category.Module.limits
+
+/-!
+# The category of `R`-modules has finite biproducts
+-/
+
+open category_theory
+open category_theory.limits
+
+open_locale big_operators
+
+universes v u
+
+namespace Module
+
+variables {R : Type u} [ring R]
+
+-- As `Module R` is preadditive, and has all limits, it automatically has biproducts.
+instance : has_binary_biproducts (Module.{v} R) :=
+has_binary_biproducts.of_has_binary_products
+
+instance : has_finite_biproducts (Module.{v} R) :=
+has_finite_biproducts.of_has_finite_products
+
+-- We now construct explicit limit data,
+-- so we can compare the biproducts to the usual unbundled constructions.
+
+/--
+Construct limit data for a binary product in `Module R`, using `Module.of R (M × N)`.
+-/
+def binary_product_limit_cone (M N : Module.{v} R) : limits.limit_cone (pair M N) :=
```

Comment:
Do we want `dsimp` lemmas for the data here?

Reply:
Yes, I've added simps lemmas now, some by hand.